lib/server.js

```diff
@@ -68,6 +68,11 @@ function createServer (options, httpHandler) {
 
         multipleBindings.call(this, server, httpHandler, options, listenOptions, () => {
           this[kState].listening = true
+
+          for (const hook of this[kHooks].onListen) {
+            hook.call(this)
```

Comment:
We need to deal with errors as suggested, but beforehand we need to define how to execute the `onListen` function - serially or parallels.

I would expect:

- all the hooks run serially
- all the async hooks are awaited
- if an error is triggered, the error is logged as a warning

_(before implementing it, let discuss it a bit)_

```suggestion
     try {
        const ret = hook.call(server)
        if (ret && typeof ret.then === 'function') {
          ret.then(done, done)
          return
        }
      } catch (error) {
        err = error
      }
```

Reply:
The behavior you described looks fine to me.

I'm not sure this is the best place to put both check for this new hook.

I also didn't know how to make it work with `done`.

Reply:
Hello @Eomm, maybe I misunderstood, do you require anything else from me? I will enjoy to make this feature getting forward with your help. 😃

Reply:
I would like feedback from @fastify/core about my comment above 👍🏼

Reply:
I expect the same.